### ISSUE TITLE
[1.15.1] Crash when leaves decays suddenly.

### ISSUE THREAD

**Issue by Gbergz** [Olafski/FastLeafDecay#27]
Title^

**Minecraft**: 1.15.1
**Forge**: 30.0.42
**Mod Version**: FastLeafDecay-v20

**Crash-Report**: https://gist.github.com/Gbergz/5a08c7c12e32a8f7191814f8290b93ec

*Any ideas?*
~ Gbergz

**Comment by Quarris** [comment]
Same thing happens with Nature's Aura (https://github.com/Ellpeck/NaturesAura/issues/78).
If you want to check for leaves blocks and then cast, should do an `instance of` check instead, and then cast it.

I wanted to check excatly what you are doing the cast for, but I couldn't find the 1.15 repo. I'd think possibly to somehow decay the leave, but that (I think) can be done with properties, which wouldn't require the cast.
Anyways, hope you can fix it \o/

**Comment by Olafski** [comment]
This should be fixed with v21, available on CurseForge. 

@Quarris I'm not quite sure what you mean with doing it with properties. If you could explain that to me, I'd be happy to change it. I used to just schedule a block tick with Minecraft 1.14, but I can't find an equivalent method for 1.15 so I just call the "decay" method (`func_225542_b_`) directly. For that I need to cast to `LeavesBlock`. If there's a better way to do it, or if there actually is a method to schedule block ticks that I couldn't find, I'm all ears. :)

**Comment by Quarris** [comment]
No, thinking about it now the property thing I mentioned doesnt make any sense. Its nice that its fixed, Ellpeck's been getting quite annoyed with the amount of reports on his github from this.
